base/src/notification-manager/notification-manager.ts

```diff
@@ -9,33 +9,55 @@
   Copyright Contributors to the Zowe Project.
 */
 
+import { ZoweNotification } from './notification';
 
-export class NotificationManager implements MVDHosting.NotificationManagerInterface {
+export class ZoweNotificationManager implements MVDHosting.ZoweNotificationManagerInterface {
   private notificationCache: any[];
-  private handlers: MVDHosting.NotificationWatcher[];
+  private handlers: MVDHosting.ZoweNotificationWatcher[];
+  private ws: WebSocket;
+  private url: string;
 
   constructor() {
-    this.notificationCache = new Array<Notification>();
-    this.handlers = new Array<MVDHosting.NotificationWatcher>();
+    this.notificationCache = new Array<ZoweNotification>();
+    this.handlers = new Array<MVDHosting.ZoweNotificationWatcher>();
   }
 
-  push(notification: Notification): void {
-    this.notificationCache.push(notification);
-    for (let i = 0; i < this.handlers.length; i++) {
-      this.handlers[i].handleMessageAdded();
+  _setURL(url: string): void {
```

Comment:
This should be made more tamper-proof. If you put a var at the top of the file, outside the class, which is false but made true when _setURL is called for the first time, then you can keep `url` locally here rather than assigning it to `this.url`, and all future calls to `_setURL` can fail due to checking that you already did setURL once.

Reply:
Update: please just change this.ws to 'let ws' so it is private.